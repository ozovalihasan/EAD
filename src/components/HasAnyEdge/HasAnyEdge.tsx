import { 
  EdgeProps, 
  getBezierEdgeCenter, 
  getBezierPath, 
  Node, 
  Position 
} from 'react-flow-renderer';

import {
  RemoveEdgeButton,
  ShowEdgeText,
  LineMarker,
  CrowsFootMarker
} from "@/components"

import { getEdgeParams } from '@/utils';
import useStore from '@/zustandStore/store';

export enum HasAnyEdgeLabel {
  HasMany = "has many",
  HasOne = "has one",
}

export interface HasAnyEdgeType extends Omit<
  EdgeProps, "sourcePosition" |"targetPosition" | "data"
> {
}

export const HasAnyEdge = ({
  id,
  source,
  target,
  sourceX,
  sourceY,
  targetX,
  targetY,
  style = {},
  label,
  selected,
}: HasAnyEdgeType) => {

  const sourceNode = useStore(store => store.nodes.find( node => node.id === source)) as Node
  const targetNode = useStore(store => store.nodes.find( node => node.id === target)) as Node
  const mouseOnEdge = useStore(store => store.mouseOnEdgeId ) === id

  if (!sourceNode || !targetNode) { return <div></div> }

  const { targetPosition, ...rest } = getEdgeParams(sourceNode, targetNode);

  const edgePath = getBezierPath({...rest, targetPosition})
  

  const [centerX, centerY] = getBezierEdgeCenter({
    sourceX,
    sourceY,
    targetX: targetX,
    targetY: targetY,
  });

  
  let orient: string = "0deg" 
  if (targetPosition === Position.Bottom) {
    orient = "180deg"
  } else if (targetPosition === Position.Left) {
    orient = "-90deg"
  } else if (targetPosition === Position.Right) {
    orient = "90deg"
  } else if (targetPosition === Position.Top) {
    orient = "0deg"
  }

  let MarkerDefs = null

  if (label === HasAnyEdgeLabel.HasMany){
    MarkerDefs = <CrowsFootMarker orient={orient} edgeId={id} />
  } else if (label === HasAnyEdgeLabel.HasOne) {
    MarkerDefs = <LineMarker orient={orient} edgeId={id} />
  }
  
  return (
    <>
      {MarkerDefs}
      
      <path
        id={id}
        style={(selected || mouseOnEdge) ? {stroke: "black", strokeWidth: 3, strokeDasharray: 0} : style}
        className="stroke-first-500 fill-[none] stroke-[2] "
        d={edgePath}
        markerEnd={`url(#marker-def-${id})`}
      />

      <path
        id={`${id}-wider`}
        className="fill-[none] stroke-[15] stroke-transparent"
        d={edgePath}
      />
      { selected && 
        <foreignObject
          className='h-10 w-10'
          x={centerX - 20 / 2}
          y={centerY + 10 / 2}
        >
          <RemoveEdgeButton edgeId={id}/>
        </foreignObject> 
      }

      <ShowEdgeText label={label} centerX={centerX} centerY={centerY} />
    </>
  );
}
