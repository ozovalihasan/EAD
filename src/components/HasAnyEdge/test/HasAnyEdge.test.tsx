import { HasAnyEdge, HasAnyEdgeLabel } from '../HasAnyEdge';
import { render, screen } from "@testing-library/react";
import useStore from '@/zustandStore/store';
import useCustomizationStore from '@/zustandStore/customizationStore';

import { 
  EntityNodeType
} from '@/components'

let renderReadyComponent: JSX.Element;

const nodes: EntityNodeType[] = [
  {
    "id": "1",
    "type": "entity",
    "position": {
        "x": 465,
        "y": 150
    },
    "data": {
        "tableId": "1",
        "name": "Physician"
    },
    "width": 138,
    "height": 45,
    "selected": false,
  },
  {
    "id": "2",
    "type": "entity",
    "position": {
        "x": 585,
        "y": 255
    },
    "data": {
        "tableId": "2",
        "name": "Appointment"
    },
    "width": 138,
    "height": 45,
    "selected": false,
  },
]

beforeEach(() => {
  
  useStore.setState({ 
    nodes: nodes
  })
  
  useCustomizationStore.setState({ 
    showTextOnEdges: false
  })

  renderReadyComponent = (
    <svg>
      <HasAnyEdge
        id={"111"}
        source={"1"}
        target={"2"}
        sourceX={nodes[0].position.x}
        sourceY={nodes[0].position.y}
        targetX={nodes[1].position.x}
        targetY={nodes[1].position.y}
        label={HasAnyEdgeLabel.HasOne}
        selected={false}
      />
    </svg>
  );
});

describe('<HasAnyEdge />', () => {

  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent );
    expect(renderedContainer).toMatchSnapshot();
  });
  
  describe("if the mouse hovers on the edge", () => {
    beforeEach(()=> {
      useStore.setState({ 
        mouseOnEdgeId: "111"
      })
    })

    it('update styles', () => {
      render(renderReadyComponent );

      const allPathElements = screen.getAllByText("")[0].getElementsByTagName('path')
      
      expect(allPathElements[0]).toHaveStyle({stroke: "black", strokeWidth: 3, strokeDasharray: 0});
      expect(allPathElements.length).toBe(2);
    });
  
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })
  
  describe("if the association is 'has one'", () => {
    beforeEach(()=> {
      renderReadyComponent =(
        <svg>
          <HasAnyEdge
            id={"111"}
            source={"1"}
            target={"2"}
            sourceX={nodes[0].position.x}
            sourceY={nodes[0].position.y}
            targetX={nodes[1].position.x}
            targetY={nodes[1].position.y}
            label={HasAnyEdgeLabel.HasOne}
            selected={false}
          />
        </svg>
      )
    })


    it('renders line marker', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/MockCrossMarker/i)).toBeInTheDocument();
    });

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })

  describe("if the association is 'has many'", () => {
    beforeEach(()=> {
      renderReadyComponent =(
        <svg>
          <HasAnyEdge
            id={"111"}
            source={"1"}
            target={"2"}
            sourceX={nodes[0].position.x}
            sourceY={nodes[0].position.y}
            targetX={nodes[1].position.x}
            targetY={nodes[1].position.y}
            label={HasAnyEdgeLabel.HasMany}
            selected={false}
          />
        </svg>
      )
    })


    it('renders crow"s foot marker', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/MockCrowsFootMarker/i)).toBeInTheDocument();
    });

    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })

  describe("if it is selected", () => {
    beforeEach(()=> {
      renderReadyComponent =(
        <svg>
          <HasAnyEdge
            id={"111"}
            source={"1"}
            target={"2"}
            sourceX={nodes[0].position.x}
            sourceY={nodes[0].position.y}
            targetX={nodes[1].position.x}
            targetY={nodes[1].position.y}
            label={HasAnyEdgeLabel.HasOne}
            selected={true}
          />
        </svg>
      )
    })


    it('renders another component', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/MockRemoveEdgeButton/i)).toBeInTheDocument();
    });

    it('update styles', () => {
      
      render(renderReadyComponent );
      
      const allPathElements = screen.getAllByText("")[0].getElementsByTagName('path')

      expect(allPathElements[0]).toHaveStyle({stroke: "black", strokeWidth: 3, strokeDasharray: 0});
    });
  
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })

  describe("shows an text if it is requested", () => {
    beforeEach(()=> {
      useCustomizationStore.setState({ 
        showTextOnEdges: true
      })
    })

    it('update styles', () => {
      render(renderReadyComponent );
      expect(screen.getByText(/MockShowEdgeText/i)).toBeInTheDocument();
    });
  
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })
  
});