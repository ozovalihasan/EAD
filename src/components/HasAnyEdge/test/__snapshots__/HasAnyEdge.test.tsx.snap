// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HasAnyEdge /> if it is selected renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockLineMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
        <foreignobject
          class="h-10 w-10"
          x="584.0000000000001"
          y="230"
        >
          MockRemoveEdgeButton
          edgeId: 111
        </foreignobject>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockLineMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
      <foreignobject
        class="h-10 w-10"
        x="584.0000000000001"
        y="230"
      >
        MockRemoveEdgeButton
        edgeId: 111
      </foreignobject>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HasAnyEdge /> if the association is 'has many' renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockCrowsFootMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockCrowsFootMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HasAnyEdge /> if the association is 'has one' renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockLineMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockLineMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HasAnyEdge /> if the mouse hovers on the edge renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockLineMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockLineMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HasAnyEdge /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockLineMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockLineMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HasAnyEdge /> shows an text if it is requested renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        MockLineMarker
        orient: 0deg
        edgeId: 111
        <path
          class="stroke-first-500 fill-[none] stroke-[2] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111"
          marker-end="url(#marker-def-111)"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-wider"
        />
        MockShowEdgeText
        centerX: 594.0000000000001
        centerY: 225
        label: 
        has one
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      MockLineMarker
      orient: 0deg
      edgeId: 111
      <path
        class="stroke-first-500 fill-[none] stroke-[2] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111"
        marker-end="url(#marker-def-111)"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-wider"
      />
      MockShowEdgeText
      centerX: 594.0000000000001
      centerY: 225
      label: 
      has one
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
